File: Zachary Wales Python/ClassAverage5.py
# Zachary Wales
# Period 1
# A program using a while loop that calculate
# the class average of a number of grades for a particular class.
# Due 10/12/2022


i = 1
grade = 0
added = 0
average = 0
students = int(input("How many students are in the class? "))

#Makes sure that the number of students is valid
if students <= 0:
    while students <= 0:
        print("That number of students is unacceptable")
        students = int(input("How many students are in the class? "))
        
# Makes sure all students have a valid grade
while i <= students:
    grade = float(input("Input Grade "+str(i)+" : "))
    if grade > 100:
        print("That grade is unacceptable")
    elif grade < 0:
        print("That grade is unacceptable")
    else:   
        added = added + grade
        i= i+1

# Calculates average and prints
average = added/students
print("The class average is "+str(average))    

File: Zachary Wales Java/Workspaces/Intro/src/classAverage5.java
/*
Zachary Wales, Period 1

An application that takes in 10 grades for a quiz and tells the mean
  
Due 5/22/2023
*/

import java.util.Scanner;

public class classAverage5 {
	public static void main(String[] args) {
		// Creates input1 scanner for input
		Scanner input1 = new Scanner(System.in);

		int grade = 0;
		int sum = 0;
		int counter = 0;
		
		// And = &&, Or = ||
		
		
		// Creates a while loop for all of the grades
		while(counter<10) {
			// Sequences the counter
			counter++;
			// Prints the grade request
			System.out.print("Grade "+counter+": ");
			// Gets input
			grade = input1.nextInt();
			// Sanitizes the input
			while(grade > 100 || grade < 0) {
				System.out.print("That grade is unacceptable. Please enter a grade 0-100 : ");
				grade = input1.nextInt();
			}
			// Adds the grades to sum
			sum += grade;
		}
		// Tells the mean of the 10 numbers
		System.out.print("The mean of the 10 numbers is: "+(sum/10.0));
		
		// Closes the input
		input1.close();
	}
}
